### PR TITLE
Feat/account renew

### PR DESCRIPTION
**Pull request by marie-j**
### Description of the Change

Move exchange account renew (hosted and some providers) to my services 

### Applicable Issues

MANAGER-2006

### Related 

- [x] https://github.com/ovh-ux/ovh-api-services/pull/142

### CODE REVIEW

File: src/exchange/billing/account-renew/billing-account-renew.controller.js [hunk 1]
```diff
@@ -0,0 +1,256 @@
...
+
+      this.checkForChanges();
+    }
+
+    if (this.bufferedAccounts.list.results.every(({ renewPeriod }) => renewPeriod === period)) {
+      this.buffer.periodSelectedForAll = period;
+    }
+  }
+
+  static GetPropertyNameFromPeriodName(period) {
```
Comment (JeremyDec) [inline src/exchange/billing/account-renew/billing-account-renew.controller.js]: static method should be placed at bottom of file

File: src/exchange/billing/account-renew/billing-account-renew.template.html [hunk 2]
```diff
@@ -0,0 +1,208 @@
...
+    data-wizard-on-finish="submit"
+    data-wizard-title="'exchange_update_billing_action_title' | translate">
+
+    <div data-wizard-step data-wizard-step-valid="hasChanged()">
+
+        <p data-ng-bind-html="'exchange_update_billing_periode_intro' | translate"></p>
+
+        <div class="alert alert-warning" role="alert"
+            data-ng-if="!$ctrl.loading && $ctrl.bufferedAccounts && $ctrl.bufferedAccounts.list.messages.length > 0">
+            <div>
```
Comment (JeremyDec) [inline src/exchange/billing/account-renew/billing-account-renew.template.html]: you do need this `<div>` ?

File: src/exchange/billing/account-renew/billing-account-renew.template.html [hunk 3]
```diff
@@ -0,0 +1,208 @@
...
+                    </button>
+                </div>
+                <span class="input-group-addon" data-ng-if="!$ctrl.search.value">
+                    <span class="fa fa-search" aria-hidden="true"></span>
+                </span>
+            </div>
+        </form>
+
+        <div data-ng-if="!$ctrl.exchange" class="text-center">
+                <oui-spinner></oui-spinner>
```
Comment (JeremyDec) [inline src/exchange/billing/account-renew/billing-account-renew.template.html]:
```suggestion
            <oui-spinner></oui-spinner>
```

File: src/exchange/billing/account-renew/billing-account-renew.template.html [hunk 4]
```diff
@@ -0,0 +1,208 @@
...
+        </div>
+
+        <div data-ng-if="$ctrl.exchange">
+
+            <div class="oui-datagrid-responsive-container">
```
Comment (JeremyDec) [inline src/exchange/billing/account-renew/billing-account-renew.template.html]: No `oui-datagrid` element possible ?

File: src/exchange/billing/account-renew/billing-account-renew.template.html [hunk 5]
```diff
@@ -0,0 +1,208 @@
...
+                data-pagination-server-side-function="retrieveAccounts"
+                data-pagination-server-side-paginated-stuff="getBufferedAccounts()"
+                data-pagination-server-side-table-loading="getLoading()">
+            </div>
+        </div>
+
+        <div class="alert alert-warning mt-5" role="alert"
+            data-translate="exchange_update_billing_periode_delete_warning"
+            data-ng-if="($ctrl.services.exchangeServiceInfrastructure.isHosted() || $ctrl.services.exchangeServiceInfrastructure.isProvider()) && $ctrl.services.exchangeVersion.isAfter(2010) && !$ctrl.account.partial"
+            data-ng-show="$ctrl.model.displayDeleteWarning">
```
Comment (JeremyDec) [inline src/exchange/billing/account-renew/billing-account-renew.template.html]: why not merge this with `data-ng-if` ?

File: src/exchange/billing/account-renew/billing-account-renew.template.html (hunk 4 above)
Comment (marie-j) [reply src/exchange/billing/account-renew/billing-account-renew.template.html]: Due to specific content in the `thead` it is not possible (but a revamp has been asked and will be taken care of)

File: src/exchange/billing/account-renew/billing-account-renew.controller.js [hunk 6]
```diff
@@ -0,0 +1,256 @@
...
+
+  checkForChanges() {
+    if (this.buffer.changes == null) {
+      this.buffer.changes = [];
+    }
+
+    this.model.displayDeleteWarning = false;
+
+    if (
+      _(this.bufferedAccounts).has('list.results')
```
[the remark this answers is not in the document]
Comment (marie-j) [reply src/exchange/billing/account-renew/billing-account-renew.controller.js]: I've not refactored this part but indeed 😄

File: src/exchange/billing/account-renew/billing-account-renew.template.html (hunk 4 above)
Comment (JeremyDec) [reply src/exchange/billing/account-renew/billing-account-renew.template.html]: ok :)

File: src/exchange/billing/account-renew/billing-account-renew.controller.js (hunk 6 above)
[the remark this answers is not in the document]
Comment (JeremyDec) [reply src/exchange/billing/account-renew/billing-account-renew.controller.js]: Okay :)

File: src/exchange/billing/account-renew/billing-account-renew.controller.js (hunk 1 above)
Comment (antleblanc) [reply src/exchange/billing/account-renew/billing-account-renew.controller.js]: Is there any reason of doing this?

I'm not sure that ordered methods in that way helps.

**Pull request merged**